src/main/java/com/stosik/parking/reservation/domain/ReservationFacade.java

```diff
@@ -0,0 +1,101 @@
+import com.stosik.parking.reservation.exceptions.ReservationNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Optional;
+
+@Transactional
+@Slf4j
+public class ReservationFacade
+{
+    private final ReservationRepository reservationRepository;
+    private final PriceCalculator priceCalculator;
+    private final Meter parkingMeter;
+    private final ParkingStore parkingStore;
+    private final ReservationDtoCreator reservationDtoCreator;
+    
+    ReservationFacade(ReservationRepository reservationRepository, ParkingStore parkingStore, PriceCalculator priceCalculator,
+        Meter parkingMeter, ReservationDtoCreator reservationDtoCreator)
+    {
+        this.reservationRepository = reservationRepository;
+        this.parkingStore = parkingStore;
+        this.priceCalculator = priceCalculator;
+        this.parkingMeter = parkingMeter;
+        this.reservationDtoCreator = reservationDtoCreator;
+    }
+    
+    public ReservationDto startParkmeter(CreateReservationCommand createReservationCommand)
+    {
+        return Optional
+            .ofNullable(createReservationCommand)
+            .map(parkingMeter::startReservation)
+            .map(parkingStore::save)
+            .map(reservationDtoCreator::from)
+            .orElseThrow(CreateReservationException::new);
+    }
+    
+    public ReservationDto stopParkmeter(String licenseId)
+    {
+        return parkingStore
+            .findById(licenseId)
+            .map(parkingMeter::stopReservation)
+            .map(this::calculateCost)
+            .map(parkingStore::delete)
+            .map(reservationRepository::save)
+            .map(reservationDtoCreator::from)
+            .orElseThrow(() -> new ReservationNotFoundException(licenseId));
+    }
+    
+    public BigDecimal dispendReservationTicket(Long id)
+    {
+        return reservationRepository
+            .findById(id)
+            .map(Reservation::getCost)
+            .orElseThrow(() -> new ReservationNotFoundException(String.valueOf(id)));
+    }
+    
+    public boolean checkVehicle(String licenseId)
+    {
+        return parkingStore
+            .findAll()
+            .stream()
+            .anyMatch(reservation -> hasSameLicensePlate(reservation, licenseId));
+    }
+    
+    public BigDecimal dailyTakings(Date dateToCheck)
+    {
+        LocalDate specificDay = dateToCheck
+            .toInstant()
+            .atZone(ZoneId.systemDefault()).toLocalDate();
+        
+        return reservationRepository
+            .findByDate(specificDay.getDayOfMonth(), specificDay.getMonthValue(), specificDay.getYear())
+            .stream()
+            .map(Reservation::getCost)
+            .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+    
+    private Reservation calculateCost(Reservation reservation)
```

Comment:
liczenie kosztu zmienia obiekt... to dobrze czy źle?

Reply:
Wydaje mi sie ze lepiej byloby odseparowac wedlug CQRS te dwie operacje, jedna metoda ktora zmienia stan obiektu (command) a druga ktora go wyciaga (query).